New login auth

BAANG opened the pull request:
- Created sign in widget that uses Google for the Social Login authentication.

- Converted Victoire's routing, consolidating to a more modularized method taking advantage of the sequelize package.

Victoire44 commented on routes/api-routes.js:
```diff
@@ -26,62 +23,62 @@ module.exports = function(app) {
-module.exports = router;
+    app.delete("/favorites/:id", function (req, res) {
+        console.log(req.method, `Deleting favorite for animal id ${req.params.id}`);
+        app.Favorites.destroy({
```
Why we don't use db.Favorites here?

Victoire44 commented on routes/api-routes.js:
```diff
@@ -26,62 +23,62 @@ module.exports = function(app) {
+
+    app.get("/favorites/:id", function (req, res) {
+        console.log(req.method, `Getting favorite for animal id ${req.params.id}`);
+        app.Favorite.findOne({
```
Why we don't use db.Favorites here?

Victoire44 commented on routes/api-routes.js:
```diff
@@ -26,62 +23,62 @@ module.exports = function(app) {
+    // =============================================================
+    app.post("/favorites/:id", function (req, res) {
+        console.log(req.method, `Adding favorite for animal id ${req.params.id}`);
+        db.Favorites.create({
```
Why we don't use `db.Favorite` without a `s` here?

Victoire44 commented on models/favorite.js:
```diff
@@ -1,17 +1,8 @@
+    var Favorite = sequelize.define("Favorites", {
+        animalId: DataTypes.STRING
+    });
+    // Favorite.belongsTo(Users)
```
We should probably keep this as it adds the `userId` to the table.
To test I created a user with this query:
```
INSERT into purrfect_match_DB.users (username, password, name, location, favorites, preferences, has_preferences) 
values ("toto", "user-id", "toto", "San Francisco", null, null, false);
```

BAANG commented:
Resolved the changes you mentioned above, minus the Favorite.belongsTo() function. Node was spitting back an error when I enabled that code. So we'll work on that bit soon.

BAANG commented:
Also, @Victoire44 , the reason I removed that I changed the routes to db.Favorites is because sequelize looks for the name of the database in sql. If you'd like we can amend the database to be 'Favorite' instead of 'Favorites' but at that point it's just a matter of syntax.

Victoire44 commented on models/users.js:
```diff
@@ -1,22 +1,20 @@
-});
+    // Creates a "Users" model that matches up with DB
+    var Users = sequelize.define("Users", {
+        userId: DataTypes.STRING,
```
Why did you add `userId`? I think sequelize creates the field `id` for us already.
